### PR TITLE
[CELEBORN-290] optimize flink-plugin RemoteShuffleOutputGate/RemoteShuffleResultPartition

### PR DESCRIPTION
**Pull request by zhongqiangczq**
### What changes were proposed in this pull request?
add logs
optimize RemoteShuffleOutputGate: modify the time registershuffle from setup to pushdatahandshake
optimize RemoteShuffleResultPartition: optimize close code and add new method

### CODE REVIEW

File: client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/buffer/BufferPacker.java [hunk 1]
```diff
@@ -81,16 +89,35 @@ public void process(Buffer buffer, int subIdx) throws InterruptedException {
                 buffer.asByteBuf(),
                 BufferUtils.HEADER_LENGTH_PREFIX,
                 buffer.readableBytes() - BufferUtils.HEADER_LENGTH_PREFIX);
+        logger.debug(
```
Comment (RexXiong) [inline client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/buffer/BufferPacker.java:92]: 4 parameters given but only 3 placeholder.

File: client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/buffer/BufferPacker.java [hunk 2]
```diff
@@ -81,16 +89,35 @@ public void process(Buffer buffer, int subIdx) throws InterruptedException {
...
+            currentSubIdx,
+            buffer.getDataType(),
+            buffer.getSize(),
+            buffer.readableBytes() - BufferUtils.HEADER_LENGTH_PREFIX);
+
         buffer.recycleBuffer();
       } else {
         Buffer dumpedBuffer = cachedBuffer;
         cachedBuffer = buffer;
+        logger.debug(
```
Comment (RexXiong) [inline client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/buffer/BufferPacker.java:103]: can we implements one method to reduce code for same logging pattern.

Comment (zhongqiangczq) [reply client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/buffer/BufferPacker.java]: create function to do this

**Review by FMX — APPROVED** [review]
LGTM.

**Review by RexXiong — APPROVED** [review]

**Pull request merged**